src/server/serveractiveobject.cpp

```diff
@@ -73,3 +73,21 @@ void ServerActiveObject::dumpAOMessagesToQueue(std::queue<ActiveObjectMessage> &
 		m_messages_out.pop();
 	}
 }
+
+void ServerActiveObject::markForRemoval()
+{
+	if (!m_pending_removal) {
+		if (!isGone())
```

Comment:
This is to ensure that on_deactivated is not called twice on the same entity, without going too deep into the intricacies of when and how the remove/deactivate flags interact with one another.